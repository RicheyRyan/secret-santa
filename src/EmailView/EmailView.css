.email-view{
  padding: 1em;
}

.email-view h3 {
  text-align: center;
}

.email-view-controls{
  width: 100%;
  text-align: center;
}

.email-view-controls p, .email-view-controls button {
  display: inline-block;
}

.email-view-controls p{
  margin: 0 1em 1em;
}
