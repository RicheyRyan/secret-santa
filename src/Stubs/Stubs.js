export const users = {
  items: [
    {
      guid: '7302a5c9-f12e-45e7-a376-a9f05e4b88fc',
      name: {
        first: 'Jennifer',
        last: 'Madden',
      },
      email: 'jmadser@somemail.com',
      phone: '+353823452345',
    },
    {
      guid: 'f45e4343-6dfb-4085-b249-975e8d707b2e',
      name: {
        first: 'Bob',
        last: 'Bertler',
      },
      email: 'bb6699@nowhere.com',
      phone: '+353880456445',
    },
    {
      guid: '39bb60c9-cc52-4edd-a213-a8176f2d4d1c',
      name: {
        first: 'Lisa',
        last: 'Masterson',
      },
      email: 'lisam76@hotmail.com',
      phone: '+353823423423',
    },
    {
      guid: '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35',
      name: {
        first: 'Nelson',
        last: '',
      },
      email: 'nelsonator16@themail.com',
      phone: '+353856456456',
    },
    {
      guid: '51944eee-7eb4-4ecb-99e2-981d05f85e1e',
      name: {
        first: 'Fred',
        last: 'Walsh',
      },
      email: 'fred.walsh82@gmail.com',
      phone: '+353878234567',
    },
    {
      guid: '62d2c77e-6297-43b7-864c-042ca1ee715d',
      name: {
        first: 'Geoff',
        last: 'Lanegan',
      },
      email: 'glanegan@laneganers.ie',
      phone: '+353845645645',
    },
    {
      guid: 'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b',
      name: {
        first: 'Brian',
        last: 'Sullivan',
      },
      email: 'bsull@hmail.co.uk',
      phone: '+353823456776',
    },
    {
      guid: '587bf323-5b7b-43ca-a88f-a1e9c884c7cc',
      name: {
        first: 'Lisa',
        last: 'Desmond',
      },
      email: 'ldesmond1@shhmail.com',
      phone: '+353834543543',
    },
    {
      guid: 'cd3515f2-9b52-42ea-9f3c-e6daaf65249a',
      name: {
        first: 'Derek',
        last: 'Proctor',
      },
      email: 'theproctor@zmail.ie',
      phone: '+353878634234',
    },
  ],
  byId: {
    '7302a5c9-f12e-45e7-a376-a9f05e4b88fc': {
      guid: '7302a5c9-f12e-45e7-a376-a9f05e4b88fc',
      name: {
        first: 'Jennifer',
        last: 'Madden',
      },
      email: 'jmadser@somemail.com',
      phone: '+353823452345',
    },
    'f45e4343-6dfb-4085-b249-975e8d707b2e': {
      guid: 'f45e4343-6dfb-4085-b249-975e8d707b2e',
      name: {
        first: 'Bob',
        last: 'Bertler',
      },
      email: 'bb6699@nowhere.com',
      phone: '+353880456445',
    },
    '39bb60c9-cc52-4edd-a213-a8176f2d4d1c': {
      guid: '39bb60c9-cc52-4edd-a213-a8176f2d4d1c',
      name: {
        first: 'Lisa',
        last: 'Masterson',
      },
      email: 'lisam76@hotmail.com',
      phone: '+353823423423',
    },
    '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35': {
      guid: '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35',
      name: {
        first: 'Nelson',
        last: '',
      },
      email: 'nelsonator16@themail.com',
      phone: '+353856456456',
    },
    '51944eee-7eb4-4ecb-99e2-981d05f85e1e': {
      guid: '51944eee-7eb4-4ecb-99e2-981d05f85e1e',
      name: {
        first: 'Fred',
        last: 'Walsh',
      },
      email: 'fred.walsh82@gmail.com',
      phone: '+353878234567',
    },
    '62d2c77e-6297-43b7-864c-042ca1ee715d': {
      guid: '62d2c77e-6297-43b7-864c-042ca1ee715d',
      name: {
        first: 'Geoff',
        last: 'Lanegan',
      },
      email: 'glanegan@laneganers.ie',
      phone: '+353845645645',
    },
    'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b': {
      guid: 'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b',
      name: {
        first: 'Brian',
        last: 'Sullivan',
      },
      email: 'bsull@hmail.co.uk',
      phone: '+353823456776',
    },
    '587bf323-5b7b-43ca-a88f-a1e9c884c7cc': {
      guid: '587bf323-5b7b-43ca-a88f-a1e9c884c7cc',
      name: {
        first: 'Lisa',
        last: 'Desmond',
      },
      email: 'ldesmond1@shhmail.com',
      phone: '+353834543543',
    },
    'cd3515f2-9b52-42ea-9f3c-e6daaf65249a': {
      guid: 'cd3515f2-9b52-42ea-9f3c-e6daaf65249a',
      name: {
        first: 'Derek',
        last: 'Proctor',
      },
      email: 'theproctor@zmail.ie',
      phone: '+353878634234',
    },
  },
};
export const matches = {
  '587bf323-5b7b-43ca-a88f-a1e9c884c7cc':
    '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35',
  '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35':
    '7302a5c9-f12e-45e7-a376-a9f05e4b88fc',
  '7302a5c9-f12e-45e7-a376-a9f05e4b88fc':
    '62d2c77e-6297-43b7-864c-042ca1ee715d',
  '62d2c77e-6297-43b7-864c-042ca1ee715d':
    'f45e4343-6dfb-4085-b249-975e8d707b2e',
  'f45e4343-6dfb-4085-b249-975e8d707b2e':
    'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b',
  'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b':
    '51944eee-7eb4-4ecb-99e2-981d05f85e1e',
  '51944eee-7eb4-4ecb-99e2-981d05f85e1e':
    'cd3515f2-9b52-42ea-9f3c-e6daaf65249a',
  'cd3515f2-9b52-42ea-9f3c-e6daaf65249a':
    '39bb60c9-cc52-4edd-a213-a8176f2d4d1c',
  '39bb60c9-cc52-4edd-a213-a8176f2d4d1c':
    '587bf323-5b7b-43ca-a88f-a1e9c884c7cc',
};
