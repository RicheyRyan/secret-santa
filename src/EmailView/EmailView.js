import React from 'react';
import './EmailView.css';
import Cta from '../Cta/Cta.js';
import EmailBody from '../EmailBody/EmailBody.js';
import { getUsersName } from '../Utils/user.js';

class EmailView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentIndex: 0,
    };
  }

  previousUser = () => {
    this.setState((state, props) => {
      const nextIndex = state.currentIndex - 1;
      const usersLen = props.users.items.length;
      return {
        currentIndex: nextIndex >= 0 ? nextIndex : usersLen - 1,
      };
    });
  };

  nextUser = () => {
    this.setState((state, props) => {
      const nextIndex = state.currentIndex + 1;
      const usersLen = props.users.items.length;
      return {
        currentIndex: nextIndex < usersLen ? nextIndex : 0,
      };
    });
  };

  render() {
    const { matches, toggleEmailView, users } = this.props;
    const currentUser = users.items[this.state.currentIndex];
    const matchedUserId = matches[currentUser.guid];
    const matchedUser = users.byId[matchedUserId];

    return (
      <div className="email-view">
        <Cta
          label={'Go Back To List'}
          message={'Not finished?'}
          onClick={toggleEmailView}
        />
        <h3>
          Email for: {getUsersName(currentUser)}
        </h3>
        <div className="email-view-controls">
          <button onClick={this.previousUser}>Previous</button>
          <p>Will be sent to {currentUser.email}</p>
          <button onClick={this.nextUser}>Next</button>
        </div>
        <EmailBody user={currentUser} matchedUser={matchedUser} />
        <Cta
          label={'Send All Emails'}
          message={'Look good?'}
          onClick={this.props.sendEmails}
        />
      </div>
    );
  }
}

export default EmailView;
