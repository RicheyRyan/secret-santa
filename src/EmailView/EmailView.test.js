import React from 'react';
import renderer from 'react-test-renderer';
import EmailView from './EmailView.js';
import { users, matches } from '../Stubs/Stubs.js';

it('Cta component matches snapshot', () => {
  const component = renderer.create(
    <EmailView
      sendEmails={() => {}}
      toggleEmailView={() => {}}
      users={users}
      matches={matches}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
