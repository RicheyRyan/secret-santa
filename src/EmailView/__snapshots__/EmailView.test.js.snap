exports[`test Cta component matches snapshot 1`] = `
<div
  className="email-view">
  <div
    className="cta">
    <p>
      Not finished?
    </p>
    <button
      onClick={[Function]}>
      Go Back To List
    </button>
  </div>
  <h3>
    Email for: 
    Jennifer Madden
  </h3>
  <div
    className="email-view-controls">
    <button
      onClick={[Function]}>
      Previous
    </button>
    <p>
      Will be sent to 
      jmadser@somemail.com
    </p>
    <button
      onClick={[Function]}>
      Next
    </button>
  </div>
  <div
    className="email-body">
    <p>
      Hi 
      Jennifer
      ,
    </p>
    <p>
      Welcome to Secret Santa!
    </p>
    <p>
      You\'ve been chosen to get a gift for 
      Geoff Lanegan
      .
    </p>
    <p>
      Make sure you get them something nice.
    </p>
    <p>
      All the best,
    </p>
    <p>
      The Secret Santa Team
    </p>
  </div>
  <div
    className="cta">
    <p>
      Look good?
    </p>
    <button
      onClick={[Function]}>
      Send All Emails
    </button>
  </div>
</div>
`;
